qa/rpc-tests/blockstorage.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test the block storage methods
+WARNING:
+This test may take 10 mins or more to complete
+"""
+
+import test_framework.loginit
+
+import time
+import sys
+if sys.version_info[0] < 3:
+    raise "Use Python 3"
+import logging
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from random import randint
+
+
+class BlockstorageTest (BitcoinTestFramework):
+
+    def setup_chain(self, bitcoinConfDict=None, wallets=None):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 3, bitcoinConfDict, wallets)
+
+    def setup_network(self, split=False):
+        self.nodes = []
+        # Start 3 nodes, 0 will be mining and connected to a leveldb node(1)
+        # the leveldb node will be connected to a third node running sequential(2)
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-useblockdb=0"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-useblockdb=1"]))
+        self.nodes.append(start_node(2, self.options.tmpdir, ["-useblockdb=0"]))
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 1, 2)
+        self.is_network_split = False
+        self.sync_all()
+
+    def run_test(self):
+
+        self.sync_blocks()
+        # set some params for how many blocks to generate and how many we should test to assume
+        # chains are the same across nodes
+        generatedblocks = 500
+        blockstotest = 50
+        maxtxperblock = 20
```

Comment:
Let's make a version that runs much quicker for the automated QA.  Put these constants in a class initializer:

def __init__(self, fastRun=True):
  if fastRun:
    self.generatedblocks = 50
    ... (and the other settings)
  else:
    self.generatedblocks = 500
    ...

then you can add this test to the list here:  https://github.com/BitcoinUnlimited/BitcoinUnlimited/blob/release/qa/pull-tester/rpc-tests.py#L182